src/domain/redeem/planner/plan.v2.ts

```diff
@@ -243,151 +212,94 @@ async function getManagerAssets(args: {
-  }
+  const totalCollateralAvailable = preview.collateral
+  const debtToRepay = preview.debt
 
-  const upperBound = maxCollateralAvailable
-  let attempt = debtToRepay > upperBound ? upperBound : debtToRepay
-  if (attempt <= 0n) {
-    attempt = upperBound
+  const publicClient = getPublicClient(config)
+  if (!publicClient) {
+    throw new Error('Public client unavailable for redeem plan')
   }
 
-  let previous: bigint | undefined
-  let lastRequired: bigint | undefined
-  // Reduce iteration cap and tolerate tiny oscillations to speed up sizing.
-  const MAX_ITERATIONS = 4
-  const CONVERGENCE_BPS = 1n // stop when change <= 1 bp
-  for (let i = 0; i < MAX_ITERATIONS; i++) {
-    if (attempt <= 0n) {
-      attempt = 1n
-    }
-
-    const quote = await quoteCollateralToDebt({
-      inToken: inTokenForQuote,
-      outToken: debtAsset,
-      amountIn: attempt,
-    })
-
-    if (quote.out <= 0n) {
-      throw new Error('Quote returned zero output while sizing collateral swap')
-    }
-
-    if (quote.out < debtToRepay && attempt === upperBound) {
-      throw new Error('Collateral swap output is less than debt to repay; increase slippage')
-    }
-
-    const required = mulDivCeil(debtToRepay, attempt, quote.out)
-    lastRequired = required
-    if (required > upperBound) {
-      throw new Error(
-        'Required collateral is greater than max collateral available; increase slippage',
-      )
-    }
+  // Determine the amount of collateral the user would receive if there is 0 slippage on the swap from collateral -> debt wrt
+  // the market's underlying oracle price
+  const debtInCollateralAsset = await publicClient.readContract({
+    address: lendingAdapter,
+    abi: lendingAdapterAbi,
+    functionName: 'convertDebtToCollateralAsset',
+    args: [debtToRepay],
+  })
+  const zeroSlippageCollateralForSender = totalCollateralAvailable - debtInCollateralAsset
 
-    // Stop if converged or oscillating between two adjacent values,
-    // or if relative delta is within 1 bp (buffer applied below).
-    const converged = required === attempt
-    const oscillating = typeof previous !== 'undefined' && required === previous
-    const relDeltaBps =
-      required === 0n
-        ? 0n
-        : ((required > attempt ? required - attempt : attempt - required) * 10_000n) / required
-    if (converged || oscillating || relDeltaBps <= CONVERGENCE_BPS) {
-      return { required: applyRequiredBuffer({ required, maxCollateralAvailable: upperBound }) }
-    }
+  // Apply slippage to the amount of collateral the user would receive if there is 0 slippage on the swap from collateral -> debt wrt
+  const minCollateralForSender = calculateMinCollateralForSender(zeroSlippageCollateralForSender, slippageBps)
 
-    previous = attempt
-    attempt = required
-  }
+  // We simply use the full amount at our disposal for the swap, considering allowed slippage
+  const collateralAvailableForSwap = totalCollateralAvailable - minCollateralForSender
```

Comment:
just double checking here: this would overspend collateral cuz we are using the slippaged value as the floor? but we are fine with giving user debt asset and overspending?